lib/TH/TH.h

```diff
@@ -3,6 +3,10 @@
 
 #include "THGeneral.h"
 
+#ifdef TH_GENERIC_USE_HALF
```

Comment:
Also, to be clear: TH_GENERIC_USE_HALF is meant as temporary flag to let us convert Torch modules one at a time. So that modules that do not define it, will not break. Once we have converted all the modules (that do care about generics and use GenerateAllTypes.h from TH), the flag could be retired.